src/checkers/inference/BaseInferrableChecker.java

```diff
@@ -0,0 +1,79 @@
+package checkers.inference;
+
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.GenericAnnotatedTypeFactory;
+import org.checkerframework.javacutil.Pair;
+
+import java.util.List;
+
+import javax.lang.model.element.VariableElement;
+
+import checkers.inference.dataflow.InferenceAnalysis;
+import checkers.inference.dataflow.InferenceTransfer;
+
+import com.sun.source.tree.Tree;
+import com.sun.source.util.Trees;
+
+/**
```

Comment:
I will update this class comment to read:

"BaseInferrableChecker is the abstract base class that all inference checkers should inherit from.  It extends BaseTypeChecker through inferenceChecker.  The important differences from BaseTypeChecker is that it provides the scaffolding for configuring the system between inference/typechecking mode.  The interface InferrableChecker enumerates the methods that are necessary to make inference work.